internal/zinc-persist/src/main/java/sbt/inc/WriteMapper.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package sbt.inc;
+
+import sbt.internal.inc.mappers.RelativeWriteMapper;
+import xsbti.compile.MiniSetup;
+import xsbti.compile.analysis.Stamp;
+
+import java.io.File;
+import java.nio.file.Path;
+
+/**
+ * Defines a writer-only mapper interface that is used by Zinc before writing
+ * the contents of the analysis files to the persistent storage.
+ *
+ * This interface is useful to make the analysis file machine-independent and
+ * allow third parties to distribute these files around.
+ */
+public interface WriteMapper extends GenericMapper {
+
+    /**
+     * Defines a mapper that writes a machine-independent analysis file.
+     *
+     * An analysis file is machine independent if all the paths are relative and no
+     * information about the machine that produced it is stored -- only information
+     * about the structure of the project from a given build tool is persisted.
+     *
+     * The mapper makes sure that the analysis file looks like if it was generated by
+     * the local machine it's executed on.
+     *
+     * <b>Important note<b>: The assumption that all paths can be made relative to a concrete
+     * position is not always correct. There is no guarantee that the paths of the caches, the
+     * artifacts, the classpath entries, the compilation inputs and outputs, et cetera share
+     * the same prefix. If this is not the case this reader will fail at runtime.
+     *
+     * Such assumption is broken in our test infrastructure: `lib_managed` does not share
```

Comment:
Where do you think this should go? It has to go in a public place, otherwise Java users will not read it (the docs are also in the implementation, but consumers of this API may not have access to it).